src/Utility/TypeExtensions.cs

```diff
@@ -122,8 +122,14 @@ public static object ConvertFromJson(string json)
                 retObj = psObj.BaseObject;
             }
 
-            if (retObj is Hashtable hashtable)
+            // TODO: Review this comment to see if it is still relevant.
+            // By default, the PowerShell 7.4 language worker no longer wraps the output of ConvertFromJson to try to create a
+            // case-insensitive hashtable to support deserializing JSON which may contain keys that only differ in case.
+            // For more information, please see https://github.com/Azure/azure-functions-powershell-worker/issues/909
+
+            if (isDurableClient && (retObj is Hashtable hashtable))
             {
+                // Durable data is case sensitive when it arrives from the Host. However, customers expect Durable client data to be case insensitive, therefore, we tried to convert the Ordered
```

Comment:
Nit: complete the comment. Perhaps
```suggestion
                // Durable data is case sensitive when it arrives from the Host. However, customers expect Durable client data to be case insensitive, therefore, we try to cast the OrderedHashtable to a case-insensitive Hashtable.
```

Reply:
@michaelpeng36 -- I have updated the whole comment. Please take a look and let me know what you think. Thanks.